### ISSUE TITLE
Support range-input style pseudo-elements

### ISSUE THREAD

**Issue by Mangara** [radkovo/jStyleParser#83]
When parsing the following CSS rules:

```CSS
input[type=range]::-webkit-slider-runnable-track { ... }
input[type=range]::-webkit-slider-thumb { ... }
input[type=range]::-moz-range-track { ... }
input[type=range]::-moz-range-thumb { ... }
input[type=range]::-ms-track { ... }
input[type=range]::-ms-thumb { ... }
```

The parser gives the following error:

```
[main] ERROR cz.vutbr.web.csskit.antlr4.CSSParser - PARSING pseudo ERROR | inserting INVALID_SELPART
[main] ERROR cz.vutbr.web.csskit.antlr4.CSSErrorListener - CSSErrorListener syntaxError | extraneous input '-' expecting {INVALID_SELPART, CLASSKEYWORD, HASH, ':', ',', '>', '{', '[', '~', '+', S}
[main] ERROR cz.vutbr.web.csskit.antlr4.CSSParser - PARSING selector ERROR | inserting INVALID_SELECTOR
```

Resulting in these rules being excluded from the stylesheet.

While these pseudo-elements are non-standard and not on a standards track ([according to MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/::-moz-range-thumb)), they are nonetheless used in practice ([Source 1](https://css-tricks.com/styling-cross-browser-compatible-range-inputs-css/), [Source 2](https://www.quirksmode.org/blog/archives/2015/11/styling_and_scr.html)). As such, it would be nice to have support for them.

**Comment by radkovo** [comment]
This is partially solved by #84. However, the third-party vendor-specific pseudo-classes should not be part of jStyleParser code. Therefore, I leave this open until some general solution is found.

**Comment by Mangara** [comment]
While working on this issue, I noticed several things about the handling of pseudos that could be improved:

* All pseudos are implemented by a single class `PseudoPage`. This seems to be a reference to the pseudo-classes that modify the `@page` rule. It may make more sense to split this into two classes: `PseudoClass` and `PseudoElement`. This distinction could be carried through as far as the grammar.
* `PseudoPage` supports public setters for its main fields, `setValue` and `setFunctionName`, that are never used.
* Not all current standardized or standard-track pseudo-classes are supported. For reference, [here is the full list](https://gist.github.com/Mangara/3950981701c5c742791e3cd0ef9eb1c7) from MDN, formatted for easy inclusion.

**Comment by radkovo** [comment]
I agree. `PseudoPage` comes from far history and you're right it should be split into two classes. The unused setters should be removed and the pseudo class list should be extended as well. I would be glad if you could take care of some of these issues, I can do the rest.

**Comment by radkovo** [comment]
Ok, this seems to be solved completely by #85 #86 and #87.
